src/vm/stackwalk.cpp

```diff
@@ -1778,7 +1778,7 @@ StackWalkAction StackFrameIterator::Filter(void)
                                 pCurrTracker = pCurrTracker->GetPreviousExceptionTracker();
                             }
 
-                            if (pCurrTracker != NULL)
+                            if ((pCurrTracker != NULL) && !pCurrTracker->IsEHClauseFilter())
```

Comment:
Do you have a repro for this issue? I would like to see what's going on here and see why we don't hit this problem on amd64. 
It looks suspicious based on the comment at line 1743 that I've added when I was implementing the Unix exception handling at the beginning of the opensourcing of .net core.